relayer/chain.go

```diff
@@ -691,6 +691,7 @@ func (c *Chain) UpgradeChain(dst *Chain, plan *upgradetypes.Plan, deposit sdk.Co
 	}
 
 	upgradedClientState := clientState.ZeroCustomFields().(*ibctmtypes.ClientState)
+	upgradedClientState.LatestHeight.RevisionHeight = uint64(plan.Height + 1)
```

Comment:
This should just be `plan.Height`

Reply:
? then it wouldn't pass the check in IBC msgs.go, the client can be updated to plan.Height

Reply:
if the plan is height 100, the client *can* be successfully updated to height 100 without proving upgrades, since plan height is committed. We need the upgraded client state to start at the next height committed after the upgrade

Reply:
oh yea you're right

Reply:
```
gaiad q ibc client consensus-state 07-tendermint-12 1-350
consensus_state:
  '@type': /ibc.lightclients.tendermint.v1.ConsensusState
  next_validators_hash: 06DCEB0D7DBD4D0ABD268B711B1D2EDA5535B7A8C5724D3CACC9C7863ACD1B9B
  root:
    hash: v2uDd8PgHY/EQr/j/wgKEuZmuwkk0wV6Tzwzup7Jt98=
  timestamp: "2020-12-17T15:34:29.723662910Z"
proof: CusECugECi5jbGllbnRzLzA3LXRlbmRlcm1pbnQtMTIvY29uc2Vuc3VzU3RhdGVzLzEtMzUwEoYBCi4vaWJjLmxpZ2h0Y2xpZW50cy50ZW5kZXJtaW50LnYxLkNvbnNlbnN1c1N0YXRlElQKDAiF9e3+BRC+8IjZAhIiCiC/a4N3w+Adj8RCv+P/CAoS5ma7CSTTBXpPPDO6nsm33xogBtzrDX29TQq9JotxGx0u2lU1t6jFck08rMnHhjrNG5saDQgBGAEgASoFAALKmAQiKwgBEicEBsqYBCCJSoSpnKTBaDImS66nXTUnBxwAIapgmF/DVIqp9XUJVyAiLQgBEgYGDsqYBCAaISDFlB/pcBmTq961ncVzyLJwjz6GbxGYgofEMhw3quF0ACIrCAESJwgWypgEIAc8kJZgA//KbxprYY2FMoKGigBfZEUnzOIa7492MKCdICItCAESBgomypgEIBohIPoZCyKXUKStIBhVv/F0haqVGEOmiMwhDgLOcL1wQ3x8IisIARInDDrKmAQgcPC6BMrMyU5vobQAJLTdFyuxi/yrCBdEacjIqia86ZQgIi0IARIGDnjKmAQgGiEgNnvkAcBtUYj9SONU6qfeQtole8jXAqiX0O5A8ImSeNIiLAgBEigQ1gHKmAQgkATvM7aPMV/rJ2IxUrg9SM63PhxFjyuUCHHpFjX1bgMgIiwIARIoErgCypgEID0XwuCCEVUWvkp9N992OzkI8a0bY7EFe36IoACA+rKEICIuCAESBxSMBcqYBCAaISCyHlMNgDuwNidnO38C+k7sAwVyA9VfMtfkxyfRmBINfwrVAQrSAQoDaWJjEiADbjRaEvSyiqGzEFT2JmO/mKnTMUUkuXG9DiRjMU/4uBoJCAEYASABKgEAIicIARIBARog+VbefwQZr0EJzBl04fE3Iwq9K4y59Sd3XuzKGogXDyIiJQgBEiEBLslEDTnZQI7Wj24UhIdOzeNJF/N7n5O+dpTGG6JLGyMiJQgBEiEB49P48vUakEU23ZwqVXp9zauG+AogLGUFqx2tOnf1rdwiJwgBEgEBGiBf24V7kMxjcn1+mKHieIt9XfDHOiYYWO7jXJhfeylMCg==
proof_height:
  revision_height: "50261"
  revision_number: "0"
root@ibc-testnet-zone:~# gaiad q ibc client consensus-state 07-tendermint-12 1-351
consensus_state:
  '@type': /ibc.lightclients.tendermint.v1.ConsensusState
  next_validators_hash: 06DCEB0D7DBD4D0ABD268B711B1D2EDA5535B7A8C5724D3CACC9C7863ACD1B9B
  root:
    hash: null
  timestamp: "2020-12-17T15:34:24.231376001Z"
proof: CvQECvEECi5jbGllbnRzLzA3LXRlbmRlcm1pbnQtMTIvY29uc2Vuc3VzU3RhdGVzLzEtMzUxEmMKLi9pYmMubGlnaHRjbGllbnRzLnRlbmRlcm1pbnQudjEuQ29uc2Vuc3VzU3RhdGUSMQoLCID17f4FEIGJqm4SABogBtzrDX29TQq9JotxGx0u2lU1t6jFck08rMnHhjrNG5saDQgBGAEgASoFAALKmAQiKwgBEicCBMqYBCDbrYUZcQbr8NZKNbVs8CXNGzkWbYVGKovuUjEqNLhZSiAiLQgBEgYECMqYBCAaISAQsPrdtyrn6INgKEiTCYlMceEA5Vq3rLcjr/9xcw0l3SIrCAESJwYOypgEIJLsoRPXsag/aCOLxqyD6hK5JLHfpDpBgSkXkqZofRlPICIrCAESJwgWypgEIAc8kJZgA//KbxprYY2FMoKGigBfZEUnzOIa7492MKCdICItCAESBgomypgEIBohIPoZCyKXUKStIBhVv/F0haqVGEOmiMwhDgLOcL1wQ3x8IisIARInDDrKmAQgcPC6BMrMyU5vobQAJLTdFyuxi/yrCBdEacjIqia86ZQgIi0IARIGDnjKmAQgGiEgNnvkAcBtUYj9SONU6qfeQtole8jXAqiX0O5A8ImSeNIiLAgBEigQ1gHKmAQgkATvM7aPMV/rJ2IxUrg9SM63PhxFjyuUCHHpFjX1bgMgIiwIARIoErgCypgEID0XwuCCEVUWvkp9N992OzkI8a0bY7EFe36IoACA+rKEICIuCAESBxSMBcqYBCAaISCyHlMNgDuwNidnO38C+k7sAwVyA9VfMtfkxyfRmBINfwrVAQrSAQoDaWJjEiADbjRaEvSyiqGzEFT2JmO/mKnTMUUkuXG9DiRjMU/4uBoJCAEYASABKgEAIicIARIBARog+VbefwQZr0EJzBl04fE3Iwq9K4y59Sd3XuzKGogXDyIiJQgBEiEBLslEDTnZQI7Wj24UhIdOzeNJF/N7n5O+dpTGG6JLGyMiJQgBEiEBqtnGmt6DmqO/suechhxq6riC3D61dMjoXtn5kbUAIJYiJwgBEgEBGiDcl+XDnVYQ0pY9kqrCAw7K+wQdhsUQjY5/E6AXmAaMkw==
proof_height:
  revision_height: "50263"
  revision_number: "0"
root@ibc-testnet-zone:~# gaiad q ibc client consensus-state 07-tendermint-12 1-349
Error: 07-tendermint-12: consensus state not found
Usage:
  gaiad query ibc client consensus-state [client-id] [height] [flags]

Examples:
gaiad query ibc client  consensus-state [client-id] [height]

Flags:
      --height int      Use a specific height to query state at (this can error if the node is pruning state)
  -h, --help            help for consensus-state
      --latest-height   return latest stored consensus state
      --node string     <host>:<port> to Tendermint RPC interface for this chain (default "tcp://localhost:26657")
  -o, --output string   Output format (text|json) (default "text")
      --prove           show proofs for the query results (default true)

Global Flags:
      --chain-id string     The network chain ID
      --home string         directory for config and data (default "/root/.gaia")
      --log_format string   The logging format (json|plain) (default "plain")
      --log_level string    The logging level (trace|debug|info|warn|error|fatal|panic) (default "info")
      --trace               print out full stack trace on errors

```

plan.Height was 350